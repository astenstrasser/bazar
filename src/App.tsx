import React from "react";
import "./App.css";
import Header from "./header/Header";

const App: React.FC = () => {
  return (
    <div className="App">
      <Header />
      <div>produto 1</div>
    </div>
  );
};

export default App;
