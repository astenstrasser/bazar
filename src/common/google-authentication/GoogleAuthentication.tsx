import React from "react";
import { useGoogleAuth } from "react-gapi-auth2";
import "./GoogleAuthentication.css";
import LoginIcon from "../../assets/login-icon.svg";
import LogoutIcon from "../../assets/logout-icon.svg";
import ActionIcon from "../action-icon/ActionIcon";

function GoogleAuthentication(): React.ReactElement {
  const { googleAuth } = useGoogleAuth();

  const handleSignOut = () => {
    googleAuth?.signOut();
    alert("Deslogado com sucesso!");
  };

  return (
    <div className="google-login">
      {googleAuth?.isSignedIn.get() == false && (
        <ActionIcon
          onClick={() => googleAuth?.signIn()}
          src={LoginIcon}
          alt={"Ícone para logar na aplicação"}
        />
      )}

      {googleAuth?.isSignedIn.get() && (
        <ActionIcon
          onClick={handleSignOut}
          src={LogoutIcon}
          alt={"Ícone para deslogar da aplicação"}
        />
      )}
    </div>
  );
}

export default GoogleAuthentication;
