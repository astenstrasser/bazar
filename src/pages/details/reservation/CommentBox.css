
.item--reservation {
  margin-top: 2rem;
}
