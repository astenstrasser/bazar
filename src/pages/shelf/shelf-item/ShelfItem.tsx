import React from "react";
import "./ShelfItem.css";
import { Product } from "../../../shared-components/types";
import { useHistory } from "react-router-dom";

function ShelfItem(product: Product): React.ReactElement {
  const history = useHistory();

  const redirectToDetails = (productId: string) => {
    history.push(`/${productId}`);
  };

  return (
    <div
      className="shelf-item"
      data-testid="shelf-item"
      onClick={() => {
        redirectToDetails(product._id);
      }}
    >
      <div className="shelf-item--header">
        <div className="shelf-item--info">
          <h1>{product.name.toUpperCase()} </h1>
          <p>{product.details}</p>
        </div>
        <div className="shelf-item--price">R$ {product.price}</div>
      </div>
      <img
        className="shelf-item--picture"
        src={product.imageUrl}
        alt={product.imageAltText}
      />
    </div>
  );
}

export default ShelfItem;
