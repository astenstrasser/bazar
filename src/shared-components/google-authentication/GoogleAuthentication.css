.action-icon {
    width: 2rem;
    cursor: pointer;
}
