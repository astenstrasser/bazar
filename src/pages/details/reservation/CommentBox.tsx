import React, { useContext, useState } from "react";
import { AuthContext } from "../../../App";

function CommentBox(): React.ReactElement {
  const [comment, setComment] = useState("");
  const { user } = useContext(AuthContext);

  return (
    <div className="item--reservation">
      <h3>Reservar</h3>
      {!user && <p> Você precisa estar logado para reservar um item!</p>}

      {user && (
        <div>
          <label htmlFor="name">Nome:</label>
          <input id="name" type="text" disabled={true} value={user.getName()} />

          <label htmlFor="email">Email:</label>
          <input
            id="email"
            type="text"
            disabled={true}
            value={user.getEmail()}
          />
          <div>
            <label htmlFor="comment">Comentário:</label>
            <textarea
              id="comment"
              value={comment}
              onChange={(event) => {
                setComment(event.target.value);
              }}
            />
          </div>
          <button
            onClick={() => {
              console.log(comment, user?.getName(), user?.getEmail());
            }}
          >
            Reservar
          </button>
        </div>
      )}
    </div>
  );
}

export default CommentBox;
