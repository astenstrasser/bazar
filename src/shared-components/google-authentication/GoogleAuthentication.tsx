import React, { useContext } from "react";
import GoogleLogin, {
  GoogleLoginResponse,
  GoogleLoginResponseOffline,
  GoogleLogout,
} from "react-google-login";
import { AuthContext } from "../../App";

import "./GoogleAuthentication.css";

interface AuthProps {
  onLoginSuccess: (
    response: GoogleLoginResponse | GoogleLoginResponseOffline
  ) => void;
  onLogoutSuccess: () => void;
}

function GoogleAuthentication({
  onLoginSuccess,
  onLogoutSuccess,
}: AuthProps): React.ReactElement {
  const { isLoggedIn } = useContext(AuthContext);
  const client_id = process.env.REACT_APP_CLIENT_ID || "";

  const onFailure = () => {
    console.log("fail");
  };

  return (
    <div className="google-login">
      {!isLoggedIn && (
        <GoogleLogin
          clientId={client_id}
          buttonText="Login"
          onSuccess={onLoginSuccess}
          onFailure={onFailure}
          cookiePolicy={"single_host_origin"}
          isSignedIn={true}
        />
      )}
      {isLoggedIn && (
        <GoogleLogout
          clientId={client_id}
          buttonText="Logout"
          onFailure={onFailure}
          onLogoutSuccess={onLogoutSuccess}
        />
      )}
    </div>
  );
}

export default GoogleAuthentication;
