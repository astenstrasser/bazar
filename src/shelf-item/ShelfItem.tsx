import React from "react";
import "./ShelfItem.css";

interface Item {
  name: string;
  price: number;
  description: string;
  image: string;
  image_alt: string;
}

function ShelfItem(props: Item): React.ReactElement {
  return (
    <div className="shelf-item">
      <div className="shelf-item--header">
        <div className="shelf-item--info">
          <h1> {props.name.toUpperCase()} </h1>
          <p>{props.description}</p>
        </div>
        <div className="shelf-item--price">R$ {props.price}</div>
      </div>
      <img
        className="shelf-item--picture"
        src={props.image}
        alt={props.image_alt}
      />
    </div>
  );
}

export default ShelfItem;
