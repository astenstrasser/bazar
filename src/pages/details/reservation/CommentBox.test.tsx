import { render } from "@testing-library/react";
import ShelfItem from "../../shelf/shelf-item/ShelfItem";
import React from "react";
import CommentBox from "./CommentBox";
import { AuthContext } from "../../../App";
import GoogleAuthentication from "../../../shared-components/google-authentication/GoogleAuthentication";

function renderWithContext(authData: any) {
  return render(
    <AuthContext.Provider value={authData}>
      <CommentBox />
    </AuthContext.Provider>
  );
}

test("renders snapshot", () => {
  const { container } = renderWithContext({
    isSignedIn: true,
    user: { getName: () => "Full Name", getEmail: () => "email@example.com" },
  });
  expect(container).toMatchSnapshot();
});

test("should fill name and email with logged user information", () => {
  const { container } = renderWithContext({
    isSignedIn: true,
    user: { getName: () => "Full Name", getEmail: () => "email@example.com" },
  });

  expect(container.querySelector("#name")).toHaveValue("Full Name");
  expect(container.querySelector("#email")).toHaveValue("email@example.com");
});
