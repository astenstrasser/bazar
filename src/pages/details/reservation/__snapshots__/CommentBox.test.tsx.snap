// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div>
  <div
    class="item--reservation"
  >
    <h3>
      Reservar
    </h3>
    <div>
      <label
        for="name"
      >
        Nome:
      </label>
      <input
        disabled=""
        id="name"
        type="text"
        value="Full Name"
      />
      <label
        for="email"
      >
        Email:
      </label>
      <input
        disabled=""
        id="email"
        type="text"
        value="email@example.com"
      />
      <div>
        <label
          for="comment"
        >
          Coment√°rio:
        </label>
        <textarea
          id="comment"
        />
      </div>
      <button>
        Reservar
      </button>
    </div>
  </div>
</div>
`;
